packages/babel-plugin-transform-react-jsx-source/src/index.js

```diff
@@ -0,0 +1,56 @@
+
+ /**
+ * This adds {fileName, lineNumber} annotations to React component definitions
+ * and to jsx tag literals.
+ *
+ *
+ * == JSX Literals ==
+ *
+ * <sometag />
+ *
+ * becomes:
+ *
+ * <sometag __source={{fileName: 'this/file.js', lineNumber: 10}}/>
```

Comment:
How about instead of placing this on props, we pass some meta data to createElement so it doesnt mess with versions of react that don't know about it?